Disable PeerSharing completely for NTNV_11-12 (#4728)
input-output-hk/ouroboros-network: ouroboros-network-api/CHANGELOG.md

### Non-breaking changes

> * Fixed Codec to disable `PeerSharing` completely with buggy NTNt versions 11 and 12.

Reviewer: ```suggestion
* Fixed handshake codec: disabled `PeerSharing` for `node-to-node` versions 11 and 12.
```
